File: software/examples/c/example_simple.c
#include <stdio.h>

#define IPCON_EXPOSE_MILLISLEEP

#include "ip_connection.h"
#include "bricklet_industrial_quad_relay.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	IndustrialQuadRelay iqr;
	industrial_quad_relay_create(&iqr, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Turn relays alternating on/off for 10 times with 100 ms delay
	int i;
	for(i = 0; i < 10; ++i) {
		millisleep(100);
		industrial_quad_relay_set_value(&iqr, 1 << 0);
		millisleep(100);
		industrial_quad_relay_set_value(&iqr, 1 << 1);
		millisleep(100);
		industrial_quad_relay_set_value(&iqr, 1 << 2);
		millisleep(100);
		industrial_quad_relay_set_value(&iqr, 1 << 3);
	}

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/javascript/ExampleSimple.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletIndustrialQuadRelay = require('Tinkerforge/BrickletIndustrialQuadRelay');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'igV';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var iqr = new BrickletIndustrialQuadRelay(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        if(error === IPConnection.ERROR_ALREADY_CONNECTED) {
            console.log('Error: Already connected');        
        }
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Turn on first two relays and turn off last two relays
        ipcon.on(IPConnection.CALLBACK_CONNECTED,
            function(connectReason) {
                iqr.setValue(3);
            }
        );   
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data', function(data) {
	    ipcon.disconnect(
            function(error) {
                if(error === IPConnection.ERROR_NOT_CONNECTED) {
                    console.log('Error: Not connected');        
                }
            }
        );
process.exit(0);
});

